### PR TITLE
Added support for callback url

### PR DESCRIPTION
**Pull request by mayankamencherla**
Added support for callback url.

### CODE REVIEW

**Comment by captn3m0** [comment]
Pushed a fix for a variable assignment issue (callbackUrl), but there are still a few errors. 

```
Notice:  Undefined variable: callbackUrl in razorpay-woocommerce/razorpay-payments.php on line 262
Stack trace:
  1. {main}() index.php:0
  2. require() index.php:17
  3. require_once() wp-blog-header.php:19
  4. include() wp-includes/template-loader.php:74
  5. get_template_part() wp-content/themes/twentyfifteen/page.php:24
  6. locate_template() wp-includes/general-template.php:167
  7. load_template() wp-includes/template.php:643
  8. require() wp-includes/template.php:686
  9. the_content() wp-content/themes/twentyfifteen/content-page.php:22
 10. apply_filters() wp-includes/post-template.php:240
 11. WP_Hook->apply_filters() wp-includes/plugin.php:203
 12. do_shortcode() wp-includes/class-wp-hook.php:298
 13. preg_replace_callback() wp-includes/shortcodes.php:223
 14. do_shortcode_tag() wp-includes/shortcodes.php:223
 15. WC_Shortcodes::checkout() wp-includes/shortcodes.php:345
 16. WC_Shortcodes::shortcode_wrapper() wp-content/plugins/woocommerce/includes/class-wc-shortcodes.php:145
 17. WC_Shortcode_Checkout::output() wp-content/plugins/woocommerce/includes/class-wc-shortcodes.php:71
 18. WC_Shortcode_Checkout::order_pay() wp-content/plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php:57
 19. do_action() wp-content/plugins/woocommerce/includes/shortcodes/class-wc-shortcode-checkout.php:157
 20. WP_Hook->do_action() wp-includes/plugin.php:453
 21. WP_Hook->apply_filters() wp-includes/class-wp-hook.php:323
 22. WC_Razorpay->receipt_page() wp-includes/class-wp-hook.php:298
 23. WC_Razorpay->generate_razorpay_form() razorpay-woocommerce/razorpay-payments.php:147
 24. WC_Razorpay->getCheckoutArguments() razorpay-woocommerce/razorpay-payments.php:228
Notice:  Undefined variable: order_id in razorpay-woocommerce/razorpay-payments.php on line 626
Stack trace:
  1. {main}() index.php:0
  2. require() index.php:17
  3. wp() wp-blog-header.php:16
  4. WP->main() wp-includes/functions.php:955
  5. WP->parse_request() wp-includes/class-wp.php:733
  6. do_action_ref_array() wp-includes/class-wp.php:386
  7. WP_Hook->do_action() wp-includes/plugin.php:515
  8. WP_Hook->apply_filters() wp-includes/class-wp-hook.php:323
  9. WC_API->handle_api_requests() wp-includes/class-wp-hook.php:298
 10. do_action() wp-content/plugins/woocommerce/includes/class-wc-api.php:96
 11. WP_Hook->do_action() wp-includes/plugin.php:453
 12. WP_Hook->apply_filters() wp-includes/class-wp-hook.php:323
 13. WC_Razorpay->check_razorpay_response() wp-includes/class-wp-hook.php:298
 14. WC_Razorpay->handlePaymentStatus() razorpay-woocommerce/razorpay-payments.php:556
Notice:  Undefined variable: razorpay_payment_id in razorpay-woocommerce/razorpay-payments.php on line 629
Stack trace:
  1. {main}() index.php:0
  2. require() index.php:17
  3. wp() wp-blog-header.php:16
  4. WP->main() wp-includes/functions.php:955
  5. WP->parse_request() wp-includes/class-wp.php:733
  6. do_action_ref_array() wp-includes/class-wp.php:386
  7. WP_Hook->do_action() wp-includes/plugin.php:515
  8. WP_Hook->apply_filters() wp-includes/class-wp-hook.php:323
  9. WC_API->handle_api_requests() wp-includes/class-wp-hook.php:298
 10. do_action() wp-content/plugins/woocommerce/includes/class-wc-api.php:96
 11. WP_Hook->do_action() wp-includes/plugin.php:453
 12. WP_Hook->apply_filters() wp-includes/class-wp-hook.php:323
 13. WC_Razorpay->check_razorpay_response() wp-includes/class-wp-hook.php:298
 14. WC_Razorpay->handlePaymentStatus() razorpay-woocommerce/razorpay-payments.php:556
```

Please ensure that you have wordpress error display enabled. This is how my `wp-config.php` looks:

```php

<?php
define( 'WP_DEBUG', true );

// Enable Debug logging to the /wp-content/debug.log file
define( 'WP_DEBUG_LOG', true );

// Disable display of errors and warnings 
define( 'WP_DEBUG_DISPLAY', true );
@ini_set( 'display_errors', 0 );

// Use dev versions of core JS and CSS files (only needed if you are modifying these core files)
define( 'SCRIPT_DEBUG', true );
```

File: razorpay-payments.php [hunk 1]
```diff
@@ -533,72 +539,124 @@ function check_razorpay_response()
...
+            {
+                $this->msg['class'] = 'error';
+                $this->msg['message'] = 'Thank you for shopping with us. However, the payment failed.';
+                $order->add_order_note("Transaction Declined: $error<br/>");
+                $order->add_order_note("Payment Failed. Please check Razorpay Dashboard. <br/> Razorpay Id: $razorpay_payment_id");
+                $order->update_status('failed');
+            }
+        }
+
+        protected function handleErrorCase($order_id, & $order)
```
Comment (captn3m0) [inline razorpay-payments.php]: take only order here as well, use the getOrderId() I just committed

File: razorpay-payments.php [hunk 2]
```diff
@@ -533,72 +539,124 @@ function check_razorpay_response()
...
-                }
-                else
-                {
-                    $this->msg['class'] = 'error';
-                    $this->msg['message'] = 'Thank you for shopping with us. However, the payment failed.';
-                    $order->add_order_note("Transaction Declined: $error<br/>");
-                    $order->add_order_note("Payment Failed. Please check Razorpay Dashboard. <br/> Razorpay Id: " . $attributes[self::RAZORPAY_PAYMENT_ID]);
-                    $order->update_status('failed');
-                }
+                $this->updateOrder($order, $success, $razorpayPaymentId);
```
Comment (captn3m0) [inline razorpay-payments.php]: updateOrder should also be called with success=false for the error case?

File: razorpay-payments.php [hunk 3]
```diff
@@ -531,72 +537,132 @@ function check_razorpay_response()
...
 
-            if ($order_id and !empty($_POST[self::RAZORPAY_PAYMENT_ID]))
-            {
-                $order = new WC_Order($order_id);
+            $order = new WC_Order($orderId);
 
-                $amount = $this->getOrderAmountAsInteger($order);
+            $razorpayPaymentId = null;
 
+            if ($orderId  and !empty($_POST[self::RAZORPAY_PAYMENT_ID]))
```
Comment (ccrims0n) [inline razorpay-payments.php]: proper brackets and use explicit checks.

**Pull request merged**